### PR TITLE
Homework 5 is done

### PR DESCRIPTION
**Pull request by Kornyshev**

### CODE REVIEW

File: src/test/java/hw1/CommonConditions.java [hunk 1]
```diff
@@ -1,19 +0,0 @@
-package hw1;
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw1/CommonConditions.java]: 1. Why did you delete files for the hw1 from the current Pull Request?
2. Why files of the hw1 exist in the current Pull Request?

File: src/test/java/hw5/pages/DifferentElementsPage.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.Select;
+
+import java.util.List;
+
+public class DifferentElementsPage extends AbstractPage {
+
+    public DifferentElementsPage(WebDriver driver) {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/pages/DifferentElementsPage.java]: What is the proper place in the class for constructor according to **Java Code Convention**?

File: src/test/java/hw5/pages/MainPage.java [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
-public class MainPage extends AbstractPage {
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
 
-    public static final String HOMEPAGE_URL = "https://jdi-testing.github.io/jdi-light/index.html";
+public class MainPage extends AbstractPage {
 
     public MainPage(WebDriver driver) {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/pages/MainPage.java]: What is the proper place for constructor according to **Java Code Convention**?

File: src/test/java/hw5/pages/UserTablePage.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class UserTablePage extends AbstractPage {
+
+    public UserTablePage(WebDriver driver) {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/pages/UserTablePage.java]: What is the proper place for constructor in class according to **Java Code Convention**?

File: src/test/java/hw5/pages/UserTablePage.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+    @FindBy(css = "table#user-table tbody tr div.user-descr")
+    private List<WebElement> allDescriptions;
+
+    @FindBy(css = "table#user-table tbody tr div.user-descr input")
+    private List<WebElement> allCheckboxes;
+
+    private WebElement findConcreteUserCardOnPage(String username) {
+        for (WebElement card : usersCards) {
+            if (card.findElements(By.tagName("th")).size() == 4) {
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/pages/UserTablePage.java]: Looks like hardcoded magic number here)

File: src/test/java/hw5/tests/steps/AssertionSteps.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    @Then("{int} Number Type Dropdowns should be displayed on Users Table on User Table page")
+    public void number_Type_Dropdowns_should_be_displayed_on_Users_Table_on_User_Table_page(
+            Integer count) {
+        int selectsCount = userTablePage.getSelectsCount();
+        assertThat(selectsCount).isEqualTo(count);
+    }
+
+    @Then("{int} Usernames should be displayed on Users Table on User Table page")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/tests/steps/AssertionSteps.java]: It will be better use **Optional text** here for the better readability in the scenarios.

https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text

File: src/test/java/hw5/tests/steps/AssertionSteps.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    public void usernames_should_be_displayed_on_Users_Table_on_User_Table_page(Integer count) {
+        int usernamesCount = userTablePage.getUsernamesCount();
+        assertThat(usernamesCount).isEqualTo(count);
+    }
+
+    @Then("{int} Description texts under images should be displayed "
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/tests/steps/AssertionSteps.java]: It will be better use **Optional text** here for the better readability in the scenarios.

https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text

File: src/test/java/hw5/tests/steps/AssertionSteps.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            + "on Users Table on User Table page")
+    public void description_texts_under_images_should_be_displayed_on_Users_Table_on_User_Table_page(
+            Integer count) {
+        int descriptionsCount = userTablePage.getDescriptionsCount();
+        assertThat(descriptionsCount).isEqualTo(count);
+    }
+
+    @Then("{int} checkboxes should be displayed on Users Table on User Table page")
```
Comment (DmitryKhodakovsky) [inline src/test/java/hw5/tests/steps/AssertionSteps.java]: It will be better use **Optional text** here for the better readability in the scenarios.

https://cucumber.io/docs/cucumber/cucumber-expressions/#optional-text

**Review by DmitryKhodakovsky — CHANGES_REQUESTED** [review]

**Comment by DmitryKhodakovsky** [comment]
Please resolve conflicts and comment

File: src/test/java/hw1/CommonConditions.java (hunk 1 above)
Comment (Kornyshev) [reply src/test/java/hw1/CommonConditions.java]: These files got here from the Master branch. And I deleted them because after the changes in the pom.xml without special plugin settings (with default settings), they began to run everything that seems like a test, I decided that it was better to leave only those related to the task, everything is still in the Master branch. :)

File: src/test/java/hw5/pages/MainPage.java (hunk 3 above)
Comment (Kornyshev) [reply src/test/java/hw5/pages/MainPage.java]: Fixed

File: src/test/java/hw5/pages/DifferentElementsPage.java (hunk 2 above)
Comment (Kornyshev) [reply src/test/java/hw5/pages/DifferentElementsPage.java]: Fixed

File: src/test/java/hw5/pages/UserTablePage.java (hunk 4 above)
Comment (Kornyshev) [reply src/test/java/hw5/pages/UserTablePage.java]: Fixed

File: src/test/java/hw5/pages/UserTablePage.java (hunk 5 above)
Comment (Kornyshev) [reply src/test/java/hw5/pages/UserTablePage.java]: Fixed :)

File: src/test/java/hw5/tests/steps/AssertionSteps.java (hunk 6 above)
Comment (Kornyshev) [reply src/test/java/hw5/tests/steps/AssertionSteps.java]: Fixed

File: src/test/java/hw5/tests/steps/AssertionSteps.java (hunk 7 above)
Comment (Kornyshev) [reply src/test/java/hw5/tests/steps/AssertionSteps.java]: Fixed

File: src/test/java/hw5/tests/steps/AssertionSteps.java (hunk 8 above)
Comment (Kornyshev) [reply src/test/java/hw5/tests/steps/AssertionSteps.java]: Fixed

**Review by DmitryKhodakovsky — APPROVED** [review]
Approved. Could be merged to the **master** branch
